service/controller/clusterapi/v29/detection/detection.go

```diff
@@ -72,13 +73,10 @@ func (d *Detection) ShouldUpdate(ctx context.Context, cl v1alpha1.Cluster, md v1
 		return false, microerror.Mask(err)
 	}
 
-	// TODO(marcel):
-	/*
-		if cc.Spec.TenantCluster.AvailabilityZones != cc.Status.TenantCluster.AvailabilityZones {
-			// Add suitable logging
-			return true, nil
-		}
-	*/
+	if !reflect.DeepEqual(cc.Spec.TenantCluster.AvailabilityZones, cc.Status.TenantCluster.AvailabilityZones) {
+		d.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprint("detected the availability zones should update due to a change in the desired state of the machine deployments."))
```

Comment:
I did not have any idea how to log this. Advice is welcome.

Reply:
```suggestion
		d.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprint("detected the tenant cluster should update due to availability zone changes"))
```

Reply:
Something like this maybe, but then it would also be useful to log which AZ got added or removed. I would then do this with multiple log lines. I think trying to get everything into one line like below is not really cool.

Reply:
I was quick on the trigger finger and merged your suggestion for now.
I am not sure if there is an easy way to display the changed AZs other than the google cmp package we use in unit tests.

Reply:
I noticed you shot quickly. We can polish this later and make it pretty when someone cares. No biggy.